krystex/src/main/java/com/flipkart/krystal/krystex/node/Node.java

```diff
@@ -159,7 +183,14 @@ private void flushDecoratorsIfNeeded(DependantChain dependantChain) {
       return;
     }
     Set<RequestId> requestIds = requestsByDependantChain.get(dependantChain);
-    if (requestIds.stream().allMatch(key -> mainLogicExecuted.getOrDefault(key, false))) {
+    int requestIdExecuted = 0;
```

Comment:
Can also be written concisely as
```java
if(requestIds.stream().allMatch(key -> 
    mainLogicExecuted.getOrDefault(key, false) ||
    skipLogicRequested.getOrDefault(requestId, false))){
    //...
    }
```
Your choice - which ever you feel is more readable